### ISSUE TITLE
LSTM example returning array of same values

### ISSUE THREAD

**Issue by moumine9** [tflearn/tflearn#768]
Good morning.

I'am planning to use a LSTM network to predict crimes in the city of montreal. The dataset contains geo-time informations of eahc crimes and the category associated with the crime : 
X = [Day   Hour   Lat  Long ]  ( shape = 54733 * 4 ) 
Y = [ Type of crimes ] (shape  = 54733 * 1 )

Here is the code I am using for my network :

`
def lstm(X,Y,Xtest,Ytest):
    x = np.arange(1,Ytest.shape[0]+1)
    z = []
    y = Ytest
    n = random.randint(0,100)
    options = { "dropout":0.2, "epoch":1,"optimizer":"adam","lr":0.001 }

    m_id = ("lstm_D%s_E%s_OP%s_LR%s__%s" % ( str(options["dropout"]), str(options["epoch"]), options["optimizer"], str(options["lr"]), str(n) ))

    
    print( ("Modele en cours de consrtruction : %s \n\n" % m_id) )
    
    net = tflearn.input_data(shape = [None,4,1])
    net = tflearn.lstm(net,128, return_seq = True)
    net = tflearn.dropout(net, options["dropout"] )
    net = tflearn.lstm(net,128, return_seq = False)
    net = tflearn.dropout(net, options["dropout"] )
    net = tflearn.fully_connected(net,32,activation='relu')
    net = tflearn.fully_connected(net,1,activation = 'linear')
    
    net = tflearn.regression(net, optimizer=options["optimizer"], learning_rate=options["lr"], loss='categorical_crossentropy')

    model = tflearn.DNN(net,clip_gradients=0., tensorboard_verbose=3, checkpoint_path='/tmp/tflearn_logs/',max_checkpoints=1)
    model.fit(X, Y, batch_size = 1117, show_metric = True, n_epoch=options["epoch"],snapshot_epoch=True, run_id=m_id)

    preds = model.predict(Xtest)
    preds = np.array(preds)
    print(preds)
    for i in range(0, len(preds)):
        #z.append(np.argmax(preds[i]) )
        z.append(preds[i])

    z = np.array(z)
    y = y.reshape(x.shape)
    z = z.reshape(x.shape)

    print(z.shape)
    
    plt.plot(z,color='red', label='Prediction')
    plt.plot(y,color='blue', label='Actual')
    plt.legend(loc='best')
    plt.show()`

The problem is that this network returned an array with the same values. I have 2 theory on it :
- The first one is related to the shape of the array. Somewhere in my code, I had to do a reshape  X.reshape(-1,4,1) to be able to feed the network. I have seen some examples where the array was (-1,1,4) but could not manage to feed the network wi that form.
- The second theory is about the return_seq : according to the documentation of Tflearn, when set to true or false, the network will return either a full list or the last item of the list (2D or 3D output) . I tried the option of removing the flags, but could not manage to run the network (errors with the shapes).  

Finally, I got one more question : is LSTM suitable for sequence prediction ? One of my idea is to rearrange the data so that I can predict sequences of crimes in a specific day within a specific geographic position.
